Source/Core/VideoCommon/RenderBase.cpp

```diff
@@ -890,16 +944,27 @@ void Renderer::Swap(u32 xfbAddr, u32 fbWidth, u32 fbStride, u32 fbHeight, const
       // with the loader, and it has not been unmapped yet. Force a pipeline flush to avoid this.
       g_vertex_manager->Flush();
 
-      // Draw any imgui overlays we have. Note that "draw" here means "create commands", the actual
-      // draw calls don't get issued until DrawImGui is called, which happens in SwapImpl.
-      DrawDebugText();
-      OSD::DrawMessages();
-      ImGui::Render();
+      // Render the XFB to the screen.
```

Comment:
In addition to the additional functionality for imgui, I think splitting out 'SwapImpl' into a few more specialized functions is a nice improvement.  👍